### PR TITLE
London Class 8 - Hatice Aydogan - Javascript 2 - Week 4

### PR DESCRIPTION
**Pull request by Hatice06**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 1]
```diff
@@ -1,4 +1,19 @@
-function setAlarm() {}
+function setAlarm() {
+  const timeRemainingEl = document.getElementById("timeRemaining");
+  const setAlarmInput = document.getElementById("alarmSet");
+  let takingInput = Number(setAlarmInput.value);
```
Comment (ewintram) [inline 1-mandatory/1-alarmclock/alarmclock.js:4]: well done for converting the input value to a number!

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+  timeRemainingEl.innerText = `Time Remaining: 00:${takingInput}`;
+
+  const timerId = setInterval(() => {
+    takingInput -= 1;
+    timeRemainingEl.innerText = `Time Remaining: 00:${takingInput}`;
+
+    if (takingInput === 0) {
+      clearInterval(timerId);
```
Comment (ewintram) [inline 1-mandatory/1-alarmclock/alarmclock.js:12]: well done for using setInterval and clearInterval

File: 1-mandatory/2-quotegenerator/quotes.js [hunk 3]
```diff
@@ -490,3 +490,24 @@ const quotes = [
     author: "Zig Ziglar",
   },
 ];
+
+const quoteDiv = document.getElementById("quotes");
```
Comment (ewintram) [inline 1-mandatory/2-quotegenerator/quotes.js:494]: well done for using `getElementById`

File: 1-mandatory/2-quotegenerator/quotes.js [hunk 4]
```diff
@@ -490,3 +490,24 @@ const quotes = [
...
+let ourQuote = document.createElement("p");
+
+const authorDiv = document.getElementById("author");
+let ourAuthor = document.createElement("p");
+
+const randomQuote = pickFromArray(quotes);
+
+ourQuote.innerText = `${randomQuote.quote}`;
```
Comment (ewintram) [inline 1-mandatory/2-quotegenerator/quotes.js:502]: the value of `randomQuote.quote` and `randomQuote.author` are already strings so you don't need to interpolate them

File: 1-mandatory/3-slideshow/slideshow.js [hunk 5]
```diff
@@ -1 +1,65 @@
...
+  },
+];
+
+let carouselImg = document.querySelector("#carousel-img");
+
+let i = 0; //current image
+
+function back() {
+  if (i <= 0) {
+    i = images.length;
```
Comment (ewintram) [inline 1-mandatory/3-slideshow/slideshow.js:27]: well done for handling the beginning and the end of the carousel

File: 1-mandatory/3-slideshow/slideshow.js [hunk 6]
```diff
@@ -1 +1,65 @@
...
+let fwdMove; 
+
+let autoFwdButton = document.querySelector("#auto-forward-btn");
+autoFwdButton.addEventListener("click", function () {
+fwdMove = setInterval(forward, 1000);
+});
+
+let stopButton = document.querySelector("#stop-btn");
+stopButton.addEventListener("click", () => {
+  clearInterval(backMove);
```
Comment (ewintram) [inline 1-mandatory/3-slideshow/slideshow.js:63]: well done for using clear interval

File: 1-mandatory/3-slideshow/slideshow.js [hunk 7]
```diff
@@ -1 +1,65 @@
...
+backButton.addEventListener("click", back);
+
+let fwdButton = document.querySelector("#forward-btn");
+fwdButton.addEventListener("click", forward);
+
+let backMove;
+
+let autoBackButton = document.querySelector("#auto-back-btn");
+autoBackButton.addEventListener("click", function () {
+backMove = setInterval(back, 1000);
```
Comment (ewintram) [inline 1-mandatory/3-slideshow/slideshow.js:51]: well done for creating re-usable functions for `forward` and `back` and using them the to auto slideshow!

File: 1-mandatory/2-quotegenerator/quotes.js (hunk 4 above)
Comment (Hatice06) [reply 1-mandatory/2-quotegenerator/quotes.js:502]: Thanks Ellie you're right. I deleted them.

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 8]
```diff
@@ -1,4 +1,19 @@
-function setAlarm() {}
+function setAlarm() {
+  const timeRemainingEl = document.getElementById("timeRemaining");
+  const setAlarmInput = document.getElementById("alarmSet");
+  let takingInput = Number(setAlarmInput.value);
+  timeRemainingEl.innerText = `Time Remaining: 00:${takingInput}`;
```
[the remark this answers is not in the document]
Comment (Hatice06) [reply 1-mandatory/1-alarmclock/alarmclock.js:5]: Thank you Ellie for your help in buddy group meeting.

**Pull request closed without merging**